From 5587cebb5c5551b0c64480a5d8c7907c23ffd884 Mon Sep 17 00:00:00 2001
From: chentau <tchen1998@gmail.com>
Date: Sat, 30 Jan 2021 22:35:25 -0800
Subject: [PATCH 04/12] fix some tests

---
 src/nvim/edit.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/src/nvim/edit.c b/src/nvim/edit.c
index 99fa414c5..36e5ff854 100644
--- a/src/nvim/edit.c
+++ b/src/nvim/edit.c
@@ -214,7 +214,7 @@ static bool compl_started = false;
 // Which Ctrl-X mode are we in?
 static int ctrl_x_mode = CTRL_X_NORMAL;
 
-static int compl_matches = 0;
+static int compl_matches = 0;               // number of completion items
 static char_u     *compl_pattern = NULL;
 static Direction compl_direction = FORWARD;
 static Direction compl_shows_dir = FORWARD;
@@ -2824,6 +2824,9 @@ int set_compl_match_array(void)
     lead_len = (int)STRLEN(compl_leader);
   }
 
+  if (compl_matches <= 0) {
+    return -1;
+  }
   scores = xcalloc(compl_matches + 1, sizeof(compl_score_T));
 
   do {
@@ -2874,7 +2877,7 @@ int set_compl_match_array(void)
   compl = compl_first_match;
   do {
     if ((compl->cp_flags & CP_ORIGINAL_TEXT) == 0
-        && (scores[j].score)) {
+        && (compl_leader == NULL || scores[j].score)) {
       if (!shown_match_ok) {
         if (compl == compl_shown_match || did_find_shown_match) {
           // This item is the shown match or this is the
@@ -2938,7 +2941,7 @@ int set_compl_match_array(void)
 /// Show the popup menu for the list of matches.
 void ins_compl_show_pum(void)
 {
-  int cur;
+  int cur = -1;
   int i;
   colnr_T col;
   bool array_changed = false;
@@ -5414,7 +5417,7 @@ static int ins_complete(int c, bool enable_pum)
   n = ins_compl_next(true, ins_compl_key2count(c), insert_match, false);
 
 
-  if (n > 1) {          // all matches have been found
+  if (n >= 1) {          // all matches have been found
     compl_matches = n;
   }
   compl_curr_match = compl_shown_match;
@@ -5476,7 +5479,7 @@ static int ins_complete(int c, bool enable_pum)
          * Translations may need more than twice that. */
         static char_u match_ref[81];
 
-        if (compl_matches > 0)
+        if (compl_matches >= 0)
           vim_snprintf((char *)match_ref, sizeof(match_ref),
               _("match %d of %d"),
               compl_curr_match->cp_number, compl_matches);
-- 
2.30.1

