From 5a11e392f457837de72679cdb2fc2dafcccf5c93 Mon Sep 17 00:00:00 2001
From: chentau <tchen1998@gmail.com>
Date: Tue, 2 Feb 2021 21:31:14 -0800
Subject: [PATCH 09/12] filtering now works with completefunc (hopefully)

---
 src/nvim/edit.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/src/nvim/edit.c b/src/nvim/edit.c
index ddebc0e21..89f74ef7f 100644
--- a/src/nvim/edit.c
+++ b/src/nvim/edit.c
@@ -2639,7 +2639,6 @@ void set_completion(colnr_T startcol, list_T *list)
   // compl_pattern doesn't need to be set
   compl_orig_text = vim_strnsave(get_cursor_line_ptr() + compl_col,
                                  compl_length);
-  compl_leader = (char_u *)xstrdup("");
   if (p_ic) {
     flags |= CP_ICASE;
   }
@@ -2837,9 +2836,10 @@ int set_compl_match_array(void)
   do {
     scores[i].compl = compl;
     if ((compl->cp_flags & CP_ORIGINAL_TEXT) == 0) {
-      scores[i].score = 1;
       if (compl_leader != NULL) {
         scores[i].score = ins_compl_match(compl, compl_leader, lead_len);
+      } else {
+        scores[i].score = ins_compl_match(compl, (char_u *)"", 0);
       }
 
       if (scores[i].score) {
@@ -2864,7 +2864,7 @@ int set_compl_match_array(void)
 
   // If we are doing filtering via a lua function, sort the completion
   // items according to the scores
-  if (active_filterfunc != LUA_NOREF && compl_leader != NULL) {
+  if (active_filterfunc != LUA_NOREF) {
     sort_completions(compl_matches + 1, scores);
   }
 
@@ -5380,12 +5380,6 @@ static int ins_complete(int c, bool enable_pum)
       compl_startpos.col = compl_col;
     }
 
-    // set compl_leader to be an empty string to enable completion sorting,
-    // except for when called from completefunc
-    if (ctrl_x_mode != CTRL_X_FUNCTION && compl_leader == NULL) {
-      compl_leader = (char_u *)xstrdup("");
-    }
-
     if (compl_cont_status & CONT_LOCAL)
       edit_submode = (char_u *)_(ctrl_x_msgs[CTRL_X_LOCAL_MSG]);
     else
-- 
2.30.1

