From 02ecc7c8414a34ca259e999b7d5d3c40d4be3e54 Mon Sep 17 00:00:00 2001
From: chentau <tchen1998@gmail.com>
Date: Tue, 2 Feb 2021 23:32:22 -0800
Subject: [PATCH 11/12] second try at fixing tests

---
 src/nvim/edit.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/nvim/edit.c b/src/nvim/edit.c
index eca282ef0..202d1adb4 100644
--- a/src/nvim/edit.c
+++ b/src/nvim/edit.c
@@ -160,6 +160,7 @@ typedef struct compl_score_S compl_score_T;
 struct compl_score_S {
   compl_T   *compl;
   double score;
+  int order;
 };
 
 
@@ -2763,7 +2764,7 @@ static int compare_scores(const void *a, const void *b)
   } else {
     // qsort is not stable: we don't want to flip two items that have
     // the same score.
-    return (compl_score_T *)a - (compl_score_T *)b;
+    return ((compl_score_T *)a)->order - ((compl_score_T *)b)->order;
   }
 }
 
@@ -2836,6 +2837,7 @@ int set_compl_match_array(void)
   scores = xcalloc(compl_matches + 1, sizeof(compl_score_T));
 
   do {
+    scores[i].order = i;
     scores[i].compl = compl;
     if ((compl->cp_flags & CP_ORIGINAL_TEXT) == 0) {
       if (compl_leader != NULL) {
-- 
2.30.1

